packages/web-client/app/utils/fetch-json.ts

```diff
@@ -0,0 +1,45 @@
+import { timeout } from 'ember-concurrency';
+import config from '@cardstack/web-client/config/environment';
+
+const FIRST_RETRY_DELAY = config.environment === 'test' ? 100 : 1000;
+class Storage404 extends Error {}
+
+export function isStorage404(e: Error): e is Storage404 {
+  return e instanceof Storage404;
+}
+
+/**
+ * Fetches resources stored in off-chain (S3) storage.
+ *
+ * @param resource The URL for the desired resource
+ * @param shouldRetry Whether to retry if fetching fails
+ * @returns The resource stored in off-chain storage
+ */
+export async function fetchJson(
```

Comment:
Since this is specific to off-chain storage, maybe it could be named `fetchOffChainJson`?